crates/network/src/transaction/mod.rs

```diff
@@ -1,22 +1,21 @@
-use crate::Receipt;
 use alloy_primitives::{Bytes, ChainId, Signature, B256, U256};
-use alloy_rlp::{BufMut, Decodable, Encodable};
+use alloy_rlp::{BufMut, Encodable};
+
+mod common;
+pub use common::TxKind;
 
 mod signed;
 pub use signed::Signed;
 
 /// Represents a minimal EVM transaction.
-pub trait Transaction: Encodable + Decodable + Send + Sync + 'static {
+pub trait Transaction: std::any::Any + Encodable + Send + Sync + 'static {
     /// The signature type for this transaction.
     ///
     /// This is usually [`alloy_primitives::Signature`], however, it may be different for future
     /// EIP-2718 transaction types, or in other networks. For example, in Optimism, the deposit
     /// transaction signature is the unit type `()`.
     type Signature;
 
-    /// The receipt type for this transaction.
-    type Receipt: Receipt;
-
```

Comment:
we'll bring this back per Prestwich thinking right? in that case let's leave this commented out instead of removing

Reply:
I don't know honestly, should receipts be tied to a specific transaction? Keeping this will make it impossible to have signers use dyn transaction because know that also has to be specified